src/sentry/static/sentry/app/components/tooltip2.jsx

```diff
@@ -0,0 +1,210 @@
+import styled from 'react-emotion';
+import PropTypes from 'prop-types';
+
+import {Manager, Reference, Popper} from 'react-popper';
+
+class Tooltip2 extends React.Component {
+  static propTypes = {
+    /**
+     * Disable the tooltip display entirely
+     */
+    disabled: PropTypes.bool,
+    /**
+     * The content to show in the tooltip popover
+     */
+    title: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+    /**
+     * Position for the tooltip.
+     */
+    position: PropTypes.oneOf([
+      'bottom',
+      'top',
+      'left',
+      'right',
+      'bottom-start',
+      'bottom-end',
+      'top-start',
+      'top-end',
+      'left-start',
+      'left-end',
+      'right-start',
+      'right-end',
+      'auto',
+    ]),
+    /**
+     * Set to true if your reference element is a styled component
+     * or needs to receive `innerRef`
+     */
+    isStyled: PropTypes.bool,
+    /**
+     * Additional style rules for the tooltip content.
+     */
+    popperStyle: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
+  };
+
+  static defaultProps = {
+    position: 'top',
+    isStyled: false,
+  };
+
+  constructor(props) {
+    super(props);
+
+    let portal = document.getElementById('tooltip-portal');
+    if (!portal) {
+      portal = document.createElement('div');
+      portal.setAttribute('id', 'tooltip-portal');
+      document.body.appendChild(portal);
+    }
+    this.portalEl = portal;
+  }
+
+  state = {
+    isOpen: false,
+  };
+
+  componentDidMount() {
+    this.tooltipId =
+      'tooltip_' +
+      Math.random()
+        .toString(36)
+        .substr(2, 10);
+  }
+
+  handleOpen = evt => {
+    this.setState({isOpen: true});
+  };
+
+  handleClose = evt => {
+    this.setState({isOpen: false});
+  };
+
+  render() {
+    const {disabled, children, title, position, popperStyle, isStyled} = this.props;
+    const {isOpen} = this.state;
+    if (disabled) {
+      return children;
+    }
+
+    let tip = null;
+    if (isOpen) {
+      tip = ReactDOM.createPortal(
+        <Popper placement={position}>
+          {({ref, style, placement, arrowProps}) => (
+            <TooltipContent
+              id={this.tooltipId}
+              aria-hidden={!isOpen}
+              isOpen={isOpen}
+              innerRef={ref}
+              style={style}
+              data-placement={placement}
+              popperStyle={popperStyle}
+            >
+              {title}
+              <TooltipArrow
+                innerRef={arrowProps.ref}
+                data-placement={placement}
+                style={arrowProps.style}
+              />
+            </TooltipContent>
+          )}
+        </Popper>,
+        this.portalEl
+      );
+    }
+
+    return (
+      <Manager>
+        <Reference>
+          {({ref}) => {
+            return React.cloneElement(children, {
+              'aria-describedby': this.tooltipId,
+              onFocus: this.handleOpen,
```

Comment:
Will these events be gobbled up by this component?

Reply:
It doesn't currently stop bubbling, I wanted to make the tooltips accessible by keyboard.

Reply:
I think these handlers will override the ones you try to apply to `children` itself. It seems unlikely to be the case but might be frustrating if a developer does try to do it.

i.e.

```javascript
<Tooltip2><button onFocus={() => // this will never get called} /></Tooltip2>
```

Reply:
Does react let us inspect the children element in a way where we can work around that? At one point I had added a wrapping span element instead of cloning, but I ran into problems with tooltips inside SVG elements.

Reply:
you can access it via `children.props` iirc